### ISSUE TITLE
Question: Test functions and Gauss-Lobatto integration

### ISSUE THREAD

**Issue by Fremih** [ehsankharazmi/hp-VPINNs#6]
Hello Ehsan,

[I](https://wias-berlin.de/~frerichs/?lang=1)'m a PhD student in Germany and I try to use your hp-VPINN approach for 2D convection-dominated convection-diffusion equations. I've implemented your idea by myself based on this code and what you wrote in your paper. I've got a small questions that I couldn't find an answer the last days. If I understand your paper correctly, then you are using as test functions P_{k+1}-P_{k-1}, where P_{k} is the k-th Legendre polynomial, to ensure that they are zero along the boundary of the cell(s). Furthermore, in section 4.1 you write that you are using Gauss-Lobatto integration and this is what I see in your code as well. I was wondering why you use Gauss-Lobatto and not Gauss-Legendre? Using Gauss-Lobatto seems to be strange to me. Gauss-Lobatto includes the boundary of the domain but by construction the test functions are zero there. Why is that important in contrast to Gauss-Legendre? 
Thanks a lot in advance and thank you for your research for your work! :)

Best
Derk

**Comment by tsarikahin** [comment]
Absolutely. Using Gauss-Lobatto integration is not a good idea as modified Legendre functions vanish them on the boundary, if the weak form is a function of the test function. If you have a different weak form, which excludes test function and instead takes its derivatives, then it makes sense to use Gauss-Labotto because you also enforce weak form on the boundaries as well which makes sense
